ethers-solc/src/artifacts/mod.rs

```diff
@@ -457,8 +457,57 @@ pub struct SettingsMetadata {
     /// The metadata hash can be removed from the bytecode via option "none".
     /// The other options are "ipfs" and "bzzr1".
     /// If the option is omitted, "ipfs" is used by default.
-    #[serde(default, rename = "bytecodeHash", skip_serializing_if = "Option::is_none")]
-    pub bytecode_hash: Option<String>,
+    #[serde(
+        default,
+        rename = "bytecodeHash",
+        skip_serializing_if = "Option::is_none",
+        with = "serde_helpers::display_from_str_opt"
+    )]
+    pub bytecode_hash: Option<BytecodeHash>,
+}
+
+impl From<BytecodeHash> for SettingsMetadata {
+    fn from(hash: BytecodeHash) -> Self {
+        Self { use_literal_content: None, bytecode_hash: Some(hash) }
+    }
+}
+
+#[derive(Clone, Debug, Copy, PartialEq, Eq, Serialize, Deserialize)]
+pub enum BytecodeHash {
+    Ipfs,
+    None,
+    Bzzr1,
+}
+
+impl Default for BytecodeHash {
+    fn default() -> Self {
+        // ipfs general default https://docs.soliditylang.org/en/latest/using-the-compiler.html#compiler-api
+        BytecodeHash::Ipfs
```

Comment:
Let's make it `None` and document that it's by default None?

Reply:
sg